app/views/shared/_user_menu.html.erb

```diff
@@ -0,0 +1,64 @@
+<div
+    class="w-full border-t border-gray-200/50 pt-6 pb-2 pr-2 bg-black flex flex-col gap-2"
+    data-controller="click-toggle"
+    data-click-toggle-flippable-class="hidden"
+    >
+
+    <a href="#"
+       data-action="click->click-toggle#toggleClasses:prevent"
```

Comment:
@krschacht notably, I didn't bind this as a target for the controller but instead attached the event handler here instead. That didn't occur to me before, but it's probably unnecessary to have the manual attach/detach event handler logic in clickToggle usage.

Reply:
Interesting, I never fully acknowledged those were two different ways to do things. It makes me wonder if the other places we have data-action could instead be converted to data-target? :) I know that's the opposite of what you did here, it seems like you're biasing towards just being explicit with data-action even though it has this perl expressiveness. :)

I'm not opposed to it though! It is kind of nice to have to write less javascript. I'd say let's keep what you did here, knowing that we now have two different approaches, and live with this a bit longer and then soon pick one approach. I'm not sure which is better.

One small suggestion: maybe remove :prevent here and put a preventDefault inside the toggleClasses function. That at least makes this data-action line slightly more readable.

Reply:
Interesting point -- since this is a link, the implicit action is a click, so I can remove `click->` from this binding. Putting preventDefault in the handler makes sense only if we want to always prevent default...setting the syntax complaints I have aside, it's nice to be able to make that choice on a per-binding basis. In this case I think it'll always be the case though.

I started moving in this direction while researching the `removeEventListener` and keyboard events. It felt too hard the way we've been doing it. Keyboard listeners get messy quick -- do I want Shift and Meta or hang the listener on the Body for a global handler? Now my controller has to be attached to the body?

But reading through the examples in the manual again -- they are incredibly interactive but never explicitly bind to targets in javascript, but rather bind actions in the html. So I think the Stimulus paradigm is to push as much as possible into the html side and the controllers end up being tiny, even when they're generic. 

Theres [this section in the reference on naming conventions](https://stimulus.hotwired.dev/reference/actions#naming-conventions) for actions and they explicitly recommend against handlers like "click" because it doesn't "help you reason about the behavior of a block of HTML without having to look at the controller source". Which is an interesting concept and I think I get where they're going.

It's clearly not my instinct, because I've been doing it the other way around...but I'm starting to suspect that the "stimulus way" is in this direction.